v2/snmp/manager.go

```diff
@@ -29,32 +35,181 @@ type Response struct {
 // If the function returns an error, the walk will be terminated.
 type Walker func(pdu *PDU) error
 
-// PDU defines ...
+// PDU defines an SNMP PDU.
 type PDU struct {
-	// TBD
+	RequestID   int32
+	Error       int
+	ErrorIndex  int
+	VarbindList []Varbind
 }
 
 type managerImpl struct {
-	conn   net.Conn
-	config *managerConfig
+	conn          net.Conn
+	config        *managerConfig
+	nextRequestID int32
 }
 
-func (m *managerImpl) Get(ctx context.Context, oids []string) (*Response, error) {
-	// TODO
-	return nil, nil
+type Varbind struct {
+	OID   asn1.ObjectIdentifier
+	Value interface{}
+}
+type packet struct {
+	Version     SNMPVersion
+	Community   []byte
+	RequestType asn1.RawValue
 }
 
-func (m *managerImpl) GetNext(ctx context.Context, oids []string) (*Response, error) {
-	// TODO
-	return nil, nil
+const maxInputBufferSize = 65535
+
+type messageType byte
+
+const getMessage = 0xA0
+const getNextMessage = 0xA1
+const getBulkMessage = 0xA5
+
+func (m *managerImpl) Get(ctx context.Context, oids []string) (*PDU, error) {
+	return m.executeGet(ctx, getMessage, 0, 0, oids)
 }
 
-func (m *managerImpl) GetBulk(ctx context.Context, oids []string, nonRepeaters int, maxRepetitions int) (*Response, error) {
-	// TODO
-	return nil, nil
+func (m *managerImpl) GetNext(ctx context.Context, oids []string) (*PDU, error) {
+	return m.executeGet(ctx, getNextMessage, 0, 0, oids)
+}
+
+func (m *managerImpl) GetBulk(ctx context.Context, oids []string, nonRepeaters, maxRepetitions int) (*PDU, error) {
+	return m.executeGet(ctx, getBulkMessage, nonRepeaters, maxRepetitions, oids)
 }
 
 func (m *managerImpl) GetWalk(ctx context.Context, oid string, walker Walker) error {
 	// TODO
 	return nil
 }
+
+func (m *managerImpl) executeGet(ctx context.Context, mType messageType, nonRepeaters, maxRepetitions int, oids []string) (*PDU, error) {
```

Comment:
param ordering different from above which is ctx then oids.

Reply:
Done